composer/trainer/trainer.py

```diff
@@ -50,60 +52,80 @@
 
 log = logging.getLogger(__name__)
 
+__all__ = ["Trainer"]
 
-class Trainer:
-    """Trainer for training a model with algorithms.
 
-    Can be created either with ``__init__`` or by providing a
-    :class:`~composer.trainer.TrainerHparams` object
-    (see :meth:`~composer.trainer.Trainer.create_from_hparams`).
+class Trainer:
+    """Trainer for training a models with Composer algorithms.
 
     Args:
         model (ComposerModel): The model to train.
-
         train_dataloader (DataLoader, DataSpec, or dict): The :class:`DataLoader`, :class:`DataSpec`,
             or dict of :class:`DataSpec` kwargs for the training data.
         max_duration (int, str, or Time): The maximum duration to train. Can be integer, which will be
             interpreted to be epochs, a str (e.g. '1ep', or '10ba'), or a :class:`Time` object.
-        eval_dataloader (Union[DataLoader, DataSpec, Evaluators], optional): The :class:`DataLoader`,
-            :class:`DataSpec`, :class:`Evaluators` for the evaluation data. The :class:`Evaluator`
-            class contains metrics relevant to the specific dataset. Set to ``None`` for no evaluation.
-        algorithms (List[Algorithm], optional): The algorithms to use during training.
-            (default: ``[]``)
-        optimizers: (Optimizers, optional): The optimizers.
-            (default: ``DecoupledSGDW(model.parameters(), lr=0.1)``)
-        schedulers: (Schedulers, optional): The schedulers.
-            (default: ``[CosineAnnealingLR()]``).
-        device (str or Device, optional): The device to use for training. Either `cpu` or `gpu`.
-            (default `cpu`)
+        eval_dataloader (DataLoader, DataSpec, or Tuple[Evaluator], optional): The :class:`DataLoader`,
```

Comment:
Since this entire section may often be consulted in code (vs the docs), please us global scope for sphinx as much as possible. e.g. 
:class:`~composer.trainer.ddp.DDPSyncStrategy`
->
:class:`.DDPSyncStrategy`

Or, if not possible due to duplication, use sphinx substitutions: